### PR TITLE
fixed Dockerfile

### LINKED ISSUE #1: Docker build and docker compose not working
Dockerfile is not building successfully

After running 

`docker build --file ./docker/Dockerfile --tag codedefenders/codedefenders:dev .
`

we get:

[INFO] 
[INFO] --- git-commit-id-plugin:4.9.10:revision (gather-git-infos) @ codedefenders ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  17.865 s
[INFO] Finished at: 2022-10-22T18:05:53Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal pl.project13.maven:git-commit-id-plugin:4.9.10:revision (gather-git-infos) on project codedefenders: .git directory is not found! Please specify a valid [dotGitDirectory] in your pom.xml -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
The command '/bin/sh -c mvn package -DskipTests -DskipCheckstyle' returned a non-zero code: 1

and if we remove that plugin from pom.xml we get few error messages relating to npm installs failing

### PR DESCRIPTION
**Pull request by hrvoje459**
docker compose now uses local dev build
fixed maven plugin configuration

### CODE REVIEW

File: docker/docker-compose.yml [hunk 1]
```diff
@@ -11,7 +11,8 @@ services:
     volumes:
       - dbvolume:/var/lib/mysql
   codedefenders:
-    image: "codedefenders/codedefenders:${CODEDEFENDERS_VERSION:?Please set a Codedefenders version via the .env file. Also see docs/Docker.md for more information.}"
+    #image: "codedefenders/codedefenders:${CODEDEFENDERS_VERSION:?Please set a Codedefenders version via the .env file. Also see docs/Docker.md for more information.}"
+    image: "codedefenders/codedefenders:dev"
```
Comment (ricnava00) [inline docker/docker-compose.yml]: I think this is good as is, the instructions specify how to do it correctly

File: pom.xml [hunk 2]
```diff
@@ -147,6 +147,7 @@
 						</goals>
 						<phase>initialize</phase>
 						<configuration>
+							<dotGitDirectory>${project.basedir}/.git</dotGitDirectory>
```
Comment (ricnava00) [inline pom.xml]: I didn't modify this and it worked anyway (after copying the .git folder with the Dockerfile), is this necessary?

**Review by ricnava00 — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 2 above)
Comment (hrvoje459) [reply pom.xml]: There is a slight difference between our files, I will use yours so I can revert this change

File: docker/docker-compose.yml (hunk 1 above)
Comment (hrvoje459) [reply docker/docker-compose.yml]: Ah yes, forgot it is set in .env file :+1:

**Review by ricnava00 — APPROVED** [review]

**Pull request merged**